src/main/java/org/apache/maven/plugins/checkstyle/AbstractCheckstyleReport.java

```diff
@@ -26,22 +26,24 @@
 import java.io.OutputStream;
 import java.nio.file.Path;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
 
+import com.puppycrawl.tools.checkstyle.AbstractAutomaticBean.OutputStreamOptions;
```

Comment:
@romani Is there a way to avoid this if you runtime swap from 9.x to 10.x?

Reply:
Deprecation happened on https://checkstyle.org/releasenotes.html#Release_10.9.3
https://github.com/checkstyle/checkstyle/issues/5337

AutomaticBean class is not removed yet , but will be at https://github.com/checkstyle/checkstyle/issues/12873

...to be extended ...

Reply:
Here is obviously the problem. All loggers except the SARIF one received ctor overloads: https://github.com/checkstyle/checkstyle/commit/6a36be871dd183304f76f73fa27045b59f0ea5c4#diff-068f0a96a3def356c86af089dd514f1f4cafd89d22eda77219f95cd4eafe8551 That is why for me only `SarifLogger` fails. Can you address this in a branch and I'll give a SNAPSHOT a shot (no pun intended)?

Reply:
@michael-o So basically you need us to add a ctor to `SarifLogger` which supports `AutomaticBean.OutputStreamOptions` as the 2nd parameter, and keep `OutputStream` as the first parameter?

I won't say that can't be done, but if you look at the source of these type of ctors, you will see we use some pretty simple code we use to convert `AutomaticBean.OutputStreamOptions` to `OutputStreamOptions`.
[````
    public XMLLogger(OutputStream outputStream,
                     AutomaticBean.OutputStreamOptions outputStreamOptions) {
        this(outputStream, OutputStreamOptions.valueOf(outputStreamOptions.name()));
    }
````](https://github.com/checkstyle/checkstyle/blob/e65027bba8de99b41f9caf71976c92d1519f7c1c/src/main/java/com/puppycrawl/tools/checkstyle/XMLLogger.java#L86-L89)

You could simply call `OutputStreamOptions.valueOf(outputStreamOptions.name())` to convert the deprecated options to the new options and not rely on us for anything. This would also allow the maven plugin to be backwards compatible for more CS versions as anything new we add will only be in the next release.

To be honest, we want to remove these deprecated stuff, but we have been stuck waiting on the maven plugin to update.

Reply:
@michael-o Can you also clarify why you are still trying to use a deprecated class (which was remove 15 months ago) if you are upgrading enough to see it is deprecated?

Reply:
@michael-o , please confirm that you can use `OutputStreamOptions.valueOf(outputStreamOptions.name())` workaround.
if you do not want to take this breaking changes, you can update to 10.9.2, but you update is big bump, better to migrate off some deprecations by API.

Reply:
Looking into. Let me fix CI errors first.

Reply:
Tried, this only works if you compile against 10.x, but does not work if you compile against 9.x and want to have upwards compat because of that. @rnveach I totally understand your point, but I would expect deprecated stuff to be consistent to some extend. I cannot update the plugin to 10.x at the moment.

I have raised a PR (https://github.com/checkstyle/checkstyle/pull/14954) and it works with current plugin by swapping the dep w/o changing any code.

Reply:
If we broke compile compatibility, then I don't think how you have a choice but to choose which version you will support. We continue to use maven plugin, even in our snapshot, so from our perspective we still have runtime compatibility with the old version of the plugin.

Reply:
> If we broke compile compatibility, then I don't know how you have a choice but to choose which version you will support. We continue to use maven plugin, even in our snapshot, so from our perspective we still have runtime compatibility with the old version of the plugin.

I didn't say that you broke it. You simply haven't thought about it. The PR I have provided is just a band-aid solution until I can lift the requirement. With the introductio of the `SarifLogger` in the plugin it does not work anymore. I have basically two options to make users happy:
* Hope that you will merge and release 10.18.0
* Revert the `SarifLogger`

I try to cover as many people as possible.

Reply:
With a strive to cover as much people as possible, we will force all of them to understand that plugin is not compatible with [10.9.3, 10.17.0], it is a lot of versions. People need some specific versions to match behavior in Eclipse and in Sonar.

If we want full happiness of users we need to put in plugin code that is aware of problems in Ctors and use ctor only if it exists.
We will remove deprecated class when we do major version bump or something big.
But again maybe it is better to break compatibility sooner. Breaking changes are inevitable, it is just just question on to do it sooner or later, but user pain will be the same.

As you already do huge non compatible changes, it might be good to make it clear that this version is up to checkstyle 10.9.2 
. And next version of plugin will be 10.9.3 and up.

So we will remove deprecated class and users will not notice it. So next big update for plugin will be migration to jdk17, whenever maven team decides to go this. But users will use plugin at jdk21 with all latest updates to validate fancy modern syntax of java .